src/TransitionGroup.js

```diff
@@ -87,9 +87,8 @@ class TransitionGroup extends React.Component {
   }
 
   performAppear = (key) => {
-    this.currentlyTransitioningKeys[key] = true;
-
     let component = this.childRefs[key];
+    this.currentlyTransitioningKeys[key] = component;
```

Comment:
I understand the concern, but I'm not convinced that this makes things any worse than they are at the moment. The only time we are holding on to an extra ref is in the situation when the child component does not call the callback in which case I think it is justified to keep the element alive?

I could write some tests around this , but I'm not quite clear how elements get 'accidentally' unmounted - do you mean when someone calls ```unmountComponentAtNode``` or when someone does ```element.removeChild``` directly on the DOM?

When an element does get 'accidentally' unmounted do we expect React to even call the ref function to null out the original ref?